packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -235,9 +292,41 @@ class UpdatePackagesCommand extends FlutterCommand {
       explicitDependencies: explicitDependencies,
       allDependencies: allDependencies,
       specialDependencies: specialDependencies,
-      doUpgrade: doUpgrade,
+      printPaths: forceUpgrade || isPrintPaths || isPrintTransitiveClosure || cherryPickPackage != null,
     );
 
+    final Iterable<PubspecDependency> baseDependencies;
+    if (cherryPickPackage != null) {
+      if (!allDependencies.containsKey(cherryPickPackage)) {
+        throwToolExit(
+            'Package "$cherryPickPackage" is not currently a dependency, and therefore cannot be upgraded.'
+        );
+      }
+      if (cherryPickVersion != null) {
+        globals.printStatus('Pinning package "$cherryPickPackage" to version "$cherryPickVersion"...');
+      } else {
+        globals.printStatus('Upgrading package "$cherryPickPackage"...');
+      }
+      final List<PubspecDependency> adjustedDependencies = <PubspecDependency>[];
+      for (final String package in allDependencies.keys) {
+        if (package == cherryPickPackage) {
+          PubspecDependency pubspec = allDependencies[cherryPickPackage]!;
+          if (cherryPickVersion != null) {
+            adjustedDependencies.add(pubspec.copyWith(version: cherryPickVersion));
+          } else {
+            adjustedDependencies.add(pubspec.copyWith(version: ">= ${pubspec.version}"));
```

Comment:
would we need this? I would prefer we only support cherrypicking if you know the exact version. Consider package:foo with the following versions available:

v1.0 - the current version in the latest Flutter stable
v2.0 - the latest version available in pub with breaking changes
v1.0+1 - a special hotfix release to fix a bug in the Flutter stable

If a user simply did `flutter update-packages --cherry-pick-package=foo` they may upgrade it to 2.0, rather than 1.0+1.

Reply:
sgtm, will update